### DISCUSSION TITLE
Reducing Azure Costs $ - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by GordonBeeming** [SSWConsulting/SSW.Rules.Content#5006] [General]
Let's talk about this rule
https://ssw.com.au/rules/reduce-azure-costs/

**Comment by premradhakrishnan** [comment]
Good tips. I have used different charts, tables etc and the "Scoped Area" chart is something new I learnt. Have found setting budgets and alerts also help.

**Comment by brydeno** [comment]
Often unexpected cost points to an implementation issue. Ie instead of processing each row of data once, I process all rows in the batch, once for every row in the batch or similar.

**Comment by BrookJeynes** [comment]
Good information, I can see this being a valuable tool to ensure that azure budgets and costs are managed

**Comment by matt-goldman** [comment]
Great rule and excellent tips. I particularly like Bryden's note that it can often be an implementation problem, requiring some investigation, rather than a resource issue, requiring infrastructure re-architecture. Would be great to see a case study showing an example of how we have tracked this down and made code changes to reduce costs. Pinning down the 'smoking gun' can often be the hardest part.

**Comment by leahy268** [comment]
It can be very hard to find out exactly what is costing money.  The Azure cost analysis is one tool in the arsenal.  It's great.  I just wish that it also worked during Azure sponsorship times.

**Comment by bradystroud** [comment]
Looks cool, I wonder if this is something that should be checked periodically e.g. in Sprint Reviews

**Reply by leahy268** [reply]
I check it weekly for our Tenant when we aren't in Sponsorship and I can.  If there are any issues (sudden changes) then I contact the developers to find out what is happening.

**Comment by ozdemir-mehmet** [comment]
What Wazza said. Needs to be enabled for Azure Sponsorship.

**Comment by AlexRothwell** [comment]
This rule looks like a great resource for times when you're trying to identify expensive services and cut down on spending. I don't need it currently but will remember it's here!

**Comment by jakebayliss** [comment]
Tip about viewing costs by resource and then drilling down into specific resources is really useful. Helps a lot to figure out why a resource is costing more than expected

**Comment by cvionnet** [comment]
Great pieces of advices, especially the Cost by resource view! As a newcomer to Azure, it is not always easy to find and follow the right information. Rule bookmarked :)

**Comment by christoment** [comment]
Great rules & video guys! - I have a keen interest in managing Azure's cost since I think this is the main threat in using cloud services 😊

**Comment by NadeekaJK** [comment]
Nice and short video about cost management. Something more to add would be on setting budget limitations.

**Comment by SamBWagner** [comment]
As someone who doesn't use the Azure portal very often, these short and prescriptive tips are the kinds of things I need. I just need to know the key questions to ask and where to get those answers. A perfect rule for me :)

**Comment by zacharykeeping** [comment]
Good video Bryden! These are good tips and I'll definitely be keeping these in mind in future Azure projects.

**Comment by amankumarrr** [comment]
The video provides good tips on managing costs over Azure, and it explains them simply and precisely

**Comment by christianmorfordwaitessw** [comment]
These filters are really good for finding those services that can have cost blowouts due to unexpected increased usage like storage accounts and databases

**Comment by danielmackay** [comment]
Hadn't noticed the cost-by-resource breakdown before.  Thanks for pointing this out @brydeno!

**Comment by danielmackay** [comment]
"Finding the big dogs 🐶" 

🤣

**Comment by tombui99** [comment]
These are some excellent tips! Keeping track and paying close attention to the cost of Azure is very important because it can go up super quickly, I can definitely see this tool being valuable for us devs to easily manage it.

**Comment by TobyChurches** [comment]
Although I haven't had to track the Azure costs in my current work, The video and rule were really informative! I'm sure these will become a great reference when I am looking to monitor the Azure budgets 🙂

**Comment by AntPolkanov** [comment]
Great video Bryden! The CostByResource view looks to be super useful during investigations. Do you use Cost Alerts much?

**Comment by Marxoz** [comment]
Good to know about the Cost views, but I will also like to see some tips on how to reduce costs if they are too high

**Comment by UlyssesMaclaren** [comment]
Thanks for spreading your knowledge amongst the other devs... This is a good topic for everyone to be aware of

**Comment by JackDuan123** [comment]
Morning guys!!!!

I am keen to work on Azure, but I have never work for a client on Azure since I join the company !!!!!!!!

**Comment by Geordie88** [comment]
While I found the topic interesting I can't really comment. As a designer, I am curious about what the devs think :)

**Comment by vladislav-kir** [comment]
Great video, great rule! It was very useful to learn more about Azure Cost Analysis!

**Comment by jimmy1992abc** [comment]
Thanks, Bryden, this is a great video to show you how to manage the Azure cost. 
The red boxes in the video are great and I love it, it makes me easily follow up, and the pace of the speech is great. 
I would like to see a recap with bullet points by the end of the video.

I am not a user of Azure at the moment, but I would like to expect more tips on how to figure out and reduce the redundancy cost.

**Comment by Levijj22** [comment]
Great video Bryden! Well done explaining how to manage Azure costs 👍👍

**Comment by lukecookssw** [comment]
Love Bryden's Azure videos. This one is no exception.

**Comment by AshAnil** [comment]
Nice video Bryden 🙂 Azure cost analysis is a great tool to keep track of our expenditures 💵
Great rule 👏

**Comment by steven0x51** [comment]
Great video Bryden! Although I haven't done Azure work so far, your content helps me prepare for my AZ-204 exam

**Comment by JackDevAU** [comment]
Great video! Some good tips and tricks from the Azure man himself. Always super invaluable, especially when the time comes and you need to figure out where all the money went 😅

**Comment by stephanfakossw** [comment]
Great video, great rule!

**Comment by drwharris** [comment]
Great topic. Nothing to add 👍

**Comment by piers-sinclair** [comment]
Azure costs can be a bit of a mythical beast. You can feel lost among the reeds without the proper knowledge and skills. Tools like Azure Cost Analysis are invaluable for diagnosing and fixing problems, so it is great to spread this knowledge around.

**Comment by Hona** [comment]
The cost analysis tool can be very useful, but also can allow deep dives into nothing. Using it correctly is very important, where investment cost vs payoff savings should be estimated beforehand.

**Comment by bahjatalaadelssw** [comment]
Is there automation or rules for safe guarding against spikes in resources? It's great seeing Bryden delivery content again!

**Comment by chrisschultzssw** [comment]
Super important stuff! As others have mentioned, setting up budgets and alerting is also very useful - as per: https://ssw.com.au/rules/azure-budgets/

**Comment by PennyWalker** [comment]
I learned to focus on your 3 biggest expenditures first before moving into any others, which makes sense. I like that there’s a lot of analysis tools.

**Comment by AdriT9** [comment]
I found the video informative and practical. Bryden did an excellent job of breaking down the various cost-saving strategies that are available when using Azure. The examples and scenarios that were presented were helpful.

I'd recommend this video to anyone who is looking to save money with Azure!

**Comment by Harry-Ross** [comment]
Great video Bryden! As someone who isn't that familiar with Azure yet, I thought it was a great summary of how to manage Azure costs! I thought the rule also succinctly summarised the points stated in the video perfectly.

**Comment by samsmithssw** [comment]
great insight on how to keep ontop of your azure spending, good for me to know how this can be implemented in client spending

**Comment by apforsyth** [comment]
Such a satisfying feeling when you  suddenly realise you can cut your expenses down with a click:-)

Better than finding a $20 note on the street!

**Comment by JeanThirion** [comment]
Good tips - Rule bookmarked, ready to use next time

**Comment by camillars** [comment]
Great explanation on how to cut down Azure costs, even for non-technical people... learning how to use the Azure Cost Analysis tool and how to analyse the provided data seems to be the way to go.

**Comment by jernejk** [comment]
I find Azue costs to be great even if the cost isn't a problem because it easily surfaces if a service isn't working as expected without needing to go in-depth into the analysis. For example, it's not always trivial to suspect that SQL Server works really hard on a specific if there are no errors and the page still responds in a reasonable time but you might notice a bump in cost for the day. (say Sunday)

You can use that knowledge to understand, why SQL Server is even working on Sunday as that was not expected and why it cost so much. It might surface some issues that are not related to cost.

**Comment by hveraus** [comment]
Simple! but powerful tips, and as usual, great explanation.

**Comment by AttackOnMorty** [comment]
I don't use Azure that much, but it's good to know these tips.

**Comment by jimmidier** [comment]
Awesome tips! The other useful bit on the chart is the Forecast cost. Using it as a reference can help understand future cost before resources are created.

**Comment by KristenHu** [comment]
Great explanation of these function, it's useful to see how could we cut Azure cost using forecast cost.

**Comment by AnkyChe** [comment]
Great video and tips. Azure cost analysis is useful and gives us an idea how to reduce unnecesary costs.

**Comment by Berber-Mama** [comment]
A non-dev like me can quickly understand the video, it's exciting to know the forecasting function.

**Comment by jerryluossw** [comment]
Great Tips - Let's easily understand every part of the cost, so we can easily decide how to reduce the cost.

**Comment by johnxu16** [comment]
It's exciting to learn how to control costs using Azure cost analysis.

**Comment by sylhuang** [comment]
This new rule will now be my checklist to review my Azure cost.

**Comment by SofieHong** [comment]
Excellent video!  I like the red box in the video, which helps me, unfamiliar with this interface, follow Brady's explanation. I would recommend this video to anyone who wants to learn how to manage Azure costs!

**Comment by Freego1783** [comment]
Nice Rule and Video on an important topic ! I'm not really using Azure for now but I've bookmarked this as I'm sure it will be useful in the future.

**Comment by Calvin-Lieu** [comment]
I haven't used Azure yet, but both the video and rule were clear and easy to understand.

**Comment by Ben0189** [comment]
Great Tips - I learnt how to analyze the azure cost using cost analysis tools from this CTF

**Comment by leehawkinsssw** [comment]
A handy short and simple approach to the basics of managing Azure costs. It was a great idea to suggest focusing on the top few contributors rather than trying to optimize for the sake of optimizing, when the cost of optimizing the small stuff simply doesn't justify the expensive developer's time.

**Comment by DavidAbraham082** [comment]
good advice
